src/bin/configurations/mod.rs

```diff
@@ -0,0 +1,483 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use clap::{parser::ValueSource, ArgMatches};
+use solang::{
+    codegen::{OptimizationLevel, Options},
+    file_resolver::FileResolver,
+    Target,
+};
+
+use std::{collections::HashSet, ffi::OsString, fs, path::PathBuf, process::exit};
+use toml::Table;
+mod unit_tests;
+
+// A macro for error handling.
+macro_rules! match_err {
+    ($func_name: expr) => {
+        match $func_name {
+            Ok(value) => value,
+            Err(err) => {
+                return Err(format!("{err}"));
+            }
+        }
+    };
+}
+
+// A type that defines the package to be compiled: file names, contract names, import maps and import paths.
+type Package = (
+    Vec<OsString>,
+    HashSet<String>,
+    Vec<PathBuf>,
+    Vec<(String, PathBuf)>,
+);
+
+type CompilerOutput = (
```

Comment:
Should this (and the type def below) be a struct instead?

Otherwise, when you see `package.2`, without reading this definition and the comment (the comment above should rather be a doc-comment) you'll have no idea what the second field of tuple is.